catalog/plugin/catalog-plugin-async/src/main/java/org/codice/ddf/catalog/plugin/async/api/internal/ProcessingFramework.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.plugin.async.api.internal;
+
+/**
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ *
+ * The {@code ProcessingFramework} processes all requests submitted to it. Available requests for processing
+ * are as follows:
+ * <ul>
+ *  <li>{@link ProcessCreateRequest}</li>
+ *  <li>{@link ProcessUpdateRequest}</li>
+ *  <li>{@link ProcessUpdateRequest}</li>
+ * </ul>
+ */
+public interface ProcessingFramework {
+
+    /**
+     * A property name to indicate that post processing has already been completed on the request
+     * in question.
+     */
+    String POST_PROCESS_COMPLETE = "post-process-complete";
+
+    /**
+     * Submits a {@code ProcessCreateRequest} to be processed by the {@code ProcessingFramework}.
+     *
+     * @param input the {@code ProcessCreateRequest} to be processed
+     */
+    void submit(ProcessCreateRequest input);
```

Comment:
If we use more generic and common interfaces, we may be able to simplify this interface down to something like this:

```
public interface ProcessingFramework {
    void submit(ProcessRequest<ProcessDeleteItem> input);
    void submit(ProcessRequest<ProcessResourceItem> input);
}
```

The question here is, will the implementations ever care if the `ProcessResourceItem` is a `ProcessCreateItem` or a `ProcessUpdateItem`. The current implementation doesn't and pretty much repeats the same code for both. That's not to say that the interface should be changed because of that. Just something to consider.